pkg/alertmanager/alertstore/bucketclient/bucket_client.go

```diff
@@ -58,14 +58,17 @@ type BucketAlertStore struct {
 
 	cfgProvider bucket.TenantConfigProvider
 	logger      log.Logger
+
+	fetchGrafanaCfg bool // Retrieve Grafana AM configs alongside Mimir AM configs.
```

Comment:
Bit of a nitpick, I prefer the argument being passed here (good!), but still don't like bools being passed around on the end of function calls.

I'd prefer a struct here:
```
type BucketAlertStoreConfig struct {
	// Retrieve Grafana AM configs alongside Mimir AM configs.
	FetchGrafanaConfig bool
```